twine/utils.py

```diff
@@ -99,23 +100,40 @@ def get_config(path: str = "~/.pypirc") -> Dict[str, RepositoryConfig]:
     return dict(config)
 
 
+def validate_url(repository_url: str) -> None:
+    """Validate the given url for allowed schemes and components"""
+
+    # Allowed schemes are http and https, based on whether the repository
+    # supports TLS or not, and scheme and host must be present in the URL
+    validator = (
+        rfc3986.validators.Validator()
+        .allow_schemes("http", "https")
+        .require_presence_of("scheme", "host")
+    )
+
+    url = rfc3986.uri_reference(repository_url)
+    validator.validate(url)
```

Comment:
I don't think the intermediate variable has much value here, and this follows the usage pattern in [the docs](https://rfc3986.readthedocs.io/en/latest/user/validating.html):

```suggestion
    validator.validate(rfc3986.uri_reference(url))
```

Reply:
Yes, combining both statements is a better idea, updated in latest commit